govcd/tm_common_test.go

```diff
@@ -80,6 +81,23 @@ func getOrCreateVCenter(vcd *TestVCD, check *C) (*VCenter, func()) {
 	}
 }
 
+func waitForListenerStatusConnected(v *VCenter) error {
+	for c := 0; c < 20; c++ {
+		err := v.Refresh()
+		if err != nil {
+			return fmt.Errorf("error refreshing vCenter: %s", err)
+		}
+
+		if v.VSphereVCenter.ListenerState == "CONNECTED" {
+			return nil
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+
+	return fmt.Errorf("failed waiting for listener state to become 'CONNECTED', got '%s'", v.VSphereVCenter.ListenerState)
```

Comment:
Suggesting something like this:

```suggestion
	return fmt.Errorf("waiting for listener state to become 'CONNECTED' expired after %d tries (%d seconds), got '%s'", v.VSphereVCenter.ListenerState)
```

Reply:
Changed